src/daemon/absorber.cc

```diff
@@ -124,40 +155,37 @@ bool Absorber::PrepareExtraFilesForCompiler(
   return true;
 }
 
-void Absorber::DoExecute(const base::WorkerPool& pool) {
+void Absorber::DoCheckCache(const base::WorkerPool& pool) {
   using namespace cache::string;
 
   while (!pool.IsShuttingDown()) {
-    Optional&& task = tasks_->Pop();
+    Optional&& task = cache_tasks_->Pop();
     if (!task) {
       break;
     }
 
-    if (task->first->IsClosed()) {
+    // Create a copy of pointer to connection here to be able to send responce
```

Comment:
I commited the change to `LockedQueue::Push()` so it shouldn't corrupt objects in case of failure. No need for local copy of `connection`.